iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -1030,7 +1033,7 @@ private void HandleTwinResponse(MqttApplicationMessageReceivedEventArgs received
                     {
                         Status = status,
                         Version = version,
-                        Message = Encoding.UTF8.GetString(payloadBytes), // This will only ever contain an error message which is encoded based on service contract (UTF-8).
+                        ErrorResponseMessage = JsonConvert.DeserializeObject<IotHubClientErrorResponseMessage>(Encoding.UTF8.GetString(payloadBytes)), // This will only ever contain an error message which is encoded based on service contract (UTF-8).
```

Comment:
I've made this change only for the mqtt-> patch twin flow. However, this will need to go into the amqp flow as well.

Reply:
We may need to wrap this with a try-catch block for `JsonReaderException` as the payload returned from the hub may not always be what we are expecting to receive. Similar handling on the service client side is like [this](https://github.com/Azure/azure-iot-sdk-csharp/blob/previews/v2/iothub/service/src/Exceptions/ExceptionHandlingHelper.cs#L29).

Reply:
That is a good point. 
Since we don't know (don't have any documentation) what all formats the error response could be in, how about this - for now we parse out the error message based on the format we know. If we encounter an exception we don't parse and set the ErrorCode to "Unknown". Once a customer reaches out to us about the "Unknown" use-case we can look at the error response and add that case in.